lib/clients/websocket.js

```diff
@@ -48,28 +46,64 @@ class WebsocketClient extends EventEmitter {
     this.socket = null;
   }
 
-  onOpen() {
-    this.emit('open');
-
+  _sendSubscription(
+    type,
+    productIDs = this.productIDs,
+    channels = this.channels
+  ) {
     const subscribeMessage = {
-      type: 'subscribe',
-      product_ids: this.productIDs,
-      channels: this.channels,
+      type: type,
+      channels: [...channels],
     };
 
+    if (productIDs) {
+      subscribeMessage.product_ids = [...productIDs];
+    }
+
     // Add Signature
     if (this.auth.secret) {
-      let sig = signRequest(
-        this.auth,
-        'GET',
-        this.channels ? '/users/self/verify' : '/users/self'
-      );
+      let sig = signRequest(this.auth, 'GET', '/users/self/verify');
       Object.assign(subscribeMessage, sig);
     }
 
     this.socket.send(JSON.stringify(subscribeMessage));
   }
 
+  subscribeProduct(...productIDs) {
```

Comment:
Not sure I agree with the spirit of this. It should be possible to pick and choose which products to listen to for each channel.